lib/resource-mapper.js

```diff
@@ -90,6 +90,14 @@ class ResourceMapper {
     return { path, contentType: contentType || this._defaultContentType }
   }
 
+  async _getMatchingFile (folder, filename, isIndex) {
+    const files = await this._readdir(folder)
+
+    // Find a file with the same name (minus the dollar extension)
+    return (files.find(f => this._removeDollarExtension(f) === filename ||
+      (isIndex && f.startsWith(this._indexFilename + '.') && !this._suffixes.some(suffix => f.endsWith(suffix)))))
```

Comment:
Seems wasteful to always iterate over the extensions.

How about, in the constructor, we instead use a regex to determine whether something is a control file? Then the default value is `/.(?:acl|meta)$/`, and this routine just does `this._controlFile.test(f)`.

Reply:
That sounds reasonable - I'll write tests that make sure stuff works, and then try to optimize if per your suggestions 😸